tests/test_cli.py

```diff
@@ -38,13 +38,17 @@ def __init__(self, case):
 
     def __enter__(self):
         self._raises_ctx.__enter__()
+        self.old_sys_stdout = sys.stdout
+        self.old_sys_stderr = sys.stderr
         sys.stdout = self.outstream = StringIO()
         sys.stderr = self.errstream = StringIO()
         return self
 
     def __exit__(self, *exc_info):
-        self.stdout, sys.stdout = self.outstream.getvalue(), sys.__stdout__
-        self.stderr, sys.stderr = self.errstream.getvalue(), sys.__stderr__
+        sys.stdout = self.old_sys_stdout
+        sys.stderr = self.old_sys_stderr
+        self.stdout = self.outstream.getvalue()
+        self.stderr = self.errstream.getvalue()
```

Comment:
To keep the reverse order, can you move these 2 lines first?
```python
def __enter__(self):
    step_1
    step_2
    step_3

def __exit__(self, *exc_info):
    step_3
    step_2
    step_1
```

Reply:
Done.